### DISCUSSION TITLE
EnemyPreferedAdd Crashes game

### DISCUSSION THREAD

**Discussion by nyancatec** [diesel-modding/BeardLib-Editor#733] [Q&A] [answered]
Hello! 
I'm trying to create a map that is somewhat good. Of course it's a mess because It's first map but whatever.
My problem is with the `EnemyPreferedAdd` element. For some reason it crashes my game.
Here's how it looks (Stopped recording around 2 seconds after it crashes):

https://user-images.githubusercontent.com/61796139/173341389-5eadf056-d9a0-4693-86a4-67bbc550a503.mp4

I know for a fact it's this element because I replaced every single one connected to Area Trigger (shown during playtest), and it's causing crash.
What's weirder is the fact that my Crappost map made for a friend in less than 10 minutes **works perfectly**. Well, not really due to lack of group spawns but still, there's control, anticipation and assault (Stopped recording after Assault starts):

https://user-images.githubusercontent.com/61796139/173341337-ba6e5e1f-36d7-4542-893b-6efc64c6252c.mp4

Both maps have Navigation Meshes if that's info that'd help. And yes, blockers/splitters are added as well.
My question is What makes this element work on one map while on the other it doesn't? And how to make it work? **Animations for spawning don't play a role here, both work properly when executed.**

**If needed, I'll give a map download in here (If possible, I don't use github often)**

**Comment by Xeletron** [comment]
Connecting enemy prefered to individual spawn points has never really been used or tested so I dont reccommend doing it. You will need to place down SpawnEnemyGroup elements, and distribute the groups to about 4-5 spawn points per element, then set what types of groups can spawn from it. In the end connect the spawn groups to the prefered add instead of the individual spawn points

**Comment by nyancatec** [comment]
Just tested what you said, It still crashes.
Not sure If I did something wrong or Editor doesn't like my messy map.
(You may see in recording a red text. It _was_ supposed to be plan B if I can't get it to work, but it requires assault, which can't happen without prefered add element. It was build befor I asked this question. It's turned off for now.)

https://user-images.githubusercontent.com/61796139/173349716-0e8da16c-f8c6-4ac1-b349-4d5f98759d74.mp4

**Comment by TonisGaming** [comment]
What I can offer to try is,
1: Turn whisperstate off, doesn't look like you are when heist goes loud
2: Rebuild nav
3: Check if adding new spawns works

**Comment by nyancatec** [comment]
Here, showed that whisper state turns off after going through area trigger. New enemy dummies added, you've seen how I generated new mesh.
I really have no clue what's going on.

https://user-images.githubusercontent.com/61796139/173354615-1624c1c2-2a4f-415e-a6f5-217ea60d7359.mp4

**Comment by Xeletron** [comment]
Whats the crashlog for this

**Comment by nyancatec** [comment]
Should I send a file or a text in a post?

**Comment by Xeletron** [comment]
wrong one, crash.txt is the correct one

**Comment by nyancatec** [comment]
Is it in Payday2 folder? If so I apparently don't have it, only these. I searched it in the folders. No results.
![image](https://user-images.githubusercontent.com/61796139/173357298-e2820e38-34db-4b29-a1a9-0f6954947dc6.png)

**Comment by Xeletron** [comment]
its in %localappdata%/PAYDAY 2

**Comment by nyancatec** [comment]
```
Application has crashed: C++ exception
[string "lib/managers/group_ai_states/groupaistatebesi..."]:3650: attempt to index field '_tweak_data' (a nil value)



SCRIPT STACK

_upd_assault_tasks() lib/managers/group_ai_states/groupaistatestreet.lua:395
clbk() lib/managers/group_ai_states/groupaistatestreet.lua:14
_execute_queued_task() lib/managers/enemymanager.lua:593
_update_queued_tasks() lib/managers/enemymanager.lua:608
update() lib/managers/enemymanager.lua:52
update() lib/setups/gamesetup.lua:769
update() lib/setups/networkgamesetup.lua:21
core/lib/setups/coresetup.lua:557


-------------------------------

Callstack:

         payday2_win32_release  (???)     ???                                                 
         payday2_win32_release  (???)     ???                                                 
         payday2_win32_release  (???)     ???                                                 
         payday2_win32_release  (???)     zip_get_name                                        
                         ntdll  (???)     RtlAllocateHeap                                     


-------------------------------

Current thread: Main

-------------------------------

System information:
	Application version : 1.123.108
	CPU : Intel(R) Core(TM) i5-4590T CPU @ 2.00GHz (2 cores); SSE; SSE2; SSE3; SSSE3; SSE4.1; SSE4.2
	DirectX : 12.0 
	GPU : Radeon RX 570 Series / aticfx32.dll[30.0.13023.4001]
	Language : english
	Memory :     16303MB 516KB
	OS : 6.2.9200 () 0x100-0x1 (64 bits)
	Physics : threaded
	Renderer : DX9 threaded
	Sound : Microsoft (Headphones (High Definition Audio Device))


```
Here it is. I think.

**Reply by nyancatec** [reply]
I was looking in game installation folder, not in %appdata%. I didn't even think to go there for Payday stuff.

**Comment by Xeletron** [comment, marked as answer]
Go to the AI tab, looks like you might have accidentaly set the Group AI state to 'street' which is broken in payday 2. Set it back to besiege

**Comment by nyancatec** [comment]
IT WORKS! Thank you so much!
I really wonder why There's so much things that are unused or crash your game.
I really feel embarrased from this.
